public/app/core/components/AppChrome/TopBar/InviteUserButton.test.tsx

```diff
@@ -128,45 +157,111 @@ describe('InviteUserButton', () => {
       mockMatchMedia(true);
+      // Simulate Grafana Cloud
+      mockConfig.namespace = 'stacks-12345';
     });
 
-    it('should handle URL generation errors gracefully', async () => {
-      mockGetExternalUserMngLinkUrl.mockImplementation(() => {
-        throw new Error('URL generation failed');
-      });
+    it('should show invite button when quota is not reached', () => {
+      mockUseGetCurrentOrgQuotaQuery.mockReturnValue({
+        data: [{ target: 'org_user', limit: 5, used: 2, org_id: 1 }],
+        error: undefined,
+        isLoading: false,
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      } as any);
+
+      render(<InviteUserButton />);
+
+      const button = screen.getByRole('button', { name: /invite user/i });
+      expect(button).toHaveTextContent('Invite');
+    });
+
+    it('should show upgrade button when quota is reached', () => {
+      mockUseGetCurrentOrgQuotaQuery.mockReturnValue({
+        data: [{ target: 'org_user', limit: 5, used: 5, org_id: 1 }],
+        error: undefined,
+        isLoading: false,
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      } as any);
+
+      render(<InviteUserButton />);
+
+      const button = screen.getByRole('button', { name: /upgrade to invite more users/i });
+      expect(button).toHaveTextContent('Upgrade');
+    });
+
+    it('should open upgrade URL when upgrade button is clicked', async () => {
+      mockUseGetCurrentOrgQuotaQuery.mockReturnValue({
+        data: [{ target: 'org_user', limit: 5, used: 5, org_id: 1 }],
+        error: undefined,
+        isLoading: false,
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      } as any);
 
-      const consoleSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
       const user = userEvent.setup();
 
       render(<InviteUserButton />);
 
-      // Should not crash when URL generation fails
-      await user.click(screen.getByRole('button', { name: /invite user/i }));
+      await user.click(screen.getByRole('button', { name: /upgrade to invite more users/i }));
 
-      expect(consoleSpy).toHaveBeenCalledWith('Failed to handle invite user click:', expect.any(Error));
+      expect(mockReportInteraction).toHaveBeenCalledWith('upgrade_user_button_clicked', {
+        placement: 'top_bar_right',
+      });
+      expect(mockGetUpgradeUrl).toHaveBeenCalledWith('upgrade-user-top-bar');
+      expect(mockWindowOpen).toHaveBeenCalledWith(mockUpgradeUrl, '_blank');
+    });
 
-      consoleSpy.mockRestore();
+    it('should not fetch quotas on on-prem instances', () => {
+      mockConfig.namespace = 'default'; // on-prem
+
+      render(<InviteUserButton />);
+
+      // Should skip the query
+      expect(mockUseGetCurrentOrgQuotaQuery).toHaveBeenCalledWith(undefined, {
+        skip: true,
+      });
     });
 
-    it('should handle popup blocking gracefully', async () => {
```

Comment:
I think we still need to test for pop up blocked errors. Since the catch block was removed from the onClick handler, the app might crash if the popup is blocked.